### ISSUE TITLE
Allow to selectively execute ram-heavy ops on CPU

### ISSUE THREAD

**Issue by romanngg** [google/jax#915]
It would be very convenient to sometimes manually ask JAX to execute a specific RAM-heavy operation (thinking about matrix inversion / solving linear systems) on CPU where more RAM is available, while letting the rest run on GPU for speed.

Thank you for consideration!

**Comment by romanngg** [comment]
Friendly ping - does the new `device_assignment` or `jax.device_put` address this by any chance?
https://jax.readthedocs.io/en/latest/jax.html#module-jax
I'm not sure what is the "XLA’s DeviceAssignment logic" and whether it allows to specify the CPU somehow.

Thanks!

**Comment by mattjj** [comment]
No, not yet. But @levskaya is just now starting to look at something that might help! #1211 is his current WIP PR.
